### PR TITLE
Use `shared/issueicon` template in projects

### PR DESCRIPTION
**Pull request by silverwind**
We can reuse the recently created subtemplate here. I also checked the whole templates for similar constructs, these appear to be the only one.

### CODE REVIEW

File: templates/projects/view.tmpl [hunk 1]
```diff
@@ -192,24 +192,8 @@
...
-											{{end}}
-										{{end}}
-									{{else}}
-										{{if .IsClosed}}
-											{{svg "octicon-issue-closed" 16 "text red"}}
-										{{else}}
-											{{svg "octicon-issue-opened" 16 "text green"}}
-										{{end}}
-									{{end}}
+								<span class="gt-dif gt-ac gt-vm">
```
Comment (lunny) [inline templates/projects/view.tmpl:195]: The `red`, `green` color has been removed?

File: templates/projects/view.tmpl [hunk 2]
```diff
@@ -192,24 +192,8 @@
...
+									{{template "shared/issueicon" .}}
```
Comment (delvh) [inline templates/projects/view.tmpl:196]: Interesting, I haven't noticed this refactoring at all.
Good cleanup!

File: templates/projects/view.tmpl (hunk 1 above)
Comment (delvh) [reply templates/projects/view.tmpl:195]: I guess because it doesn't matter at all?
It simply displays an icon that sets its color itself?

**Review by delvh — APPROVED** [review]

File: templates/projects/view.tmpl (hunk 1 above)
Comment (silverwind) [reply templates/projects/view.tmpl:195]: Yes, the SVG icons have `text <color>` classes and set color themselves, so this parent class is and was useless.

File: templates/projects/view.tmpl (hunk 2 above)
Comment (silverwind) [reply templates/projects/view.tmpl:196]: It came as part of https://github.com/go-gitea/gitea/pull/24812.

**Review by lunny — APPROVED** [review]

File: templates/repo/projects/view.tmpl [hunk 3]
```diff
@@ -198,24 +198,8 @@
...
-										{{end}}
-									{{else}}
-										{{if .IsClosed}}
-											{{svg "octicon-issue-closed" 16 "text red"}}
-										{{else}}
-											{{svg "octicon-issue-opened" 16 "text green"}}
-										{{end}}
-									{{end}}
+								<span class="gt-dif gt-ac gt-vm">
+									{{template "shared/issueicon" .}}
```
Comment (wxiaoguang) [inline templates/repo/projects/view.tmpl:202]: Is it better to put the container element `<span class="gt-dif gt-ac gt-vm">` into the issueicon template?

----

Or are these styles/classes are no-op and could be removed?

Comment (silverwind) [reply templates/repo/projects/view.tmpl:202]: They are almost certainly not noop. I would not put the wrapper into the template, as the wrapper is used to position the element and that depends on context of use.

**Comment by wxiaoguang** [comment]
But the `issueicon` only contains one svg icon, and as an icon, it should be `gt-dif gt-ac gt-vm` for all cases, right?

**Comment by silverwind** [comment]
> gt-dif gt-ac gt-vm

Adding those classes to the parent node in the issue list slighly misaligns the icon (the icon there is not vertically centered in its container there). I'm all for sharing more HTML, but in this case I just don't see it feasible and the template should only return the `<svg/>` for maximum flexibilty.

BTW, If ever necessary, we can add dict arguments to the template to alter behaviour like size for example.

**Pull request merged**